sklearn/tests/test_metadata_routing.py

```diff
@@ -104,7 +104,19 @@ def check_recorded_metadata(obj, method, **kwargs):
     records = getattr(obj, "_records", dict()).get(method, dict())
     assert set(kwargs.keys()) == set(records.keys())
     for key, value in kwargs.items():
-        assert records[key] is value
+        recorded_value = records[key]
+        # The following condition is used to check for the sample weight in an
+        # underlying estimator if the sample weight is taken as a subset of the
+        # original sample weight for example if it extracted based on the test
+        # indices.
```

Comment:
it's not necessarily only `sample_weight`. I think we can add an argument to `check_recorded_metadata` as `split_params=["sample_weight", ...]` and the caller decides which should be checked as a subset, and which as a whole.